File: E-Ticaret/Eticaret2Csharp/Kullanici.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Eticaret2_
{
    using System.Collections.Generic;

    public abstract class Kullanici
    {
        protected int id;
        protected string isim;
        protected string mail;
        protected List<Urun> urunler;

        public Kullanici()
        {
            urunler = new List<Urun>();
        }

        public abstract double GetToplamTutar(double fiyat);

        public string GetIsim()
        {
            return isim;
        }

        public List<Urun> GetUrunler()
        {
            return urunler;
        }

        public Urun GetUrun(int index)
        {
            return GetUrunler()[index];
        }

        public void AddUrun(Urun urun)
        {
            GetUrunler().Add(urun);
        }
    }

}

File: E-Ticaret/E-ticaret2/src/Kullanici.java

import java.util.ArrayList;
import java.util.List;

public abstract class Kullanici {
    protected int id;
    private String isim,mail;
    protected List<Urun> urunler;

    public void Kullanici(){
        urunler=new ArrayList<Urun>();
    }
    public abstract double getToplamTutar(double fiyat);
    public String getIsim(){
        return isim;
    }
    public List<Urun> getUrunler(){
        return urunler;
    }
    public void addUrun(Urun urun){
        getUrunler().add(urun);
    }
    public void OdemeYap(){

    }

}

